WordPress/gutenberg
# [RNMobile] Add support for `"multiple": false` block types

## Related PR
`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3042

## Description
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/1156 by adding logic for handing blocks that should only be used once in a single post. These "single use" blocks (such as the "More" block) have the following properties defined in their `block.json` file:

```
{
    "supports": {
	"multiple": false
    }
}
```

The changes in this PR enforce this configuration by disabling the block option if that block already exists in the post. 


## How has this been tested?
1. Open the demo app on either android or ios. The demo already has a **More** block added to the post.
2. Tap **+** to open the block menu - notice the **More** block is disabled. Verify tapping the **More** option does not work.
3. Cancel the block menu. 
4. Delete the **More** block in the post.
5. Tap **+** to open the block menu - notice the **More** block is now enabled. Tap to add this block to the post. Verify success.



## Screenshots 

![finished](https://user-images.githubusercontent.com/5810477/105111932-fc05e500-5a8f-11eb-9b7f-35799f320496.gif)


iOS: option enabled | iOS: option disabled|Android: option enabled | Android: option disabled
-- | --|-- | --
![more-enabled-ios](https://user-images.githubusercontent.com/5810477/105111937-ff996c00-5a8f-11eb-96dd-b1a3b0a4b5a7.png)|![more-disabled-ios](https://user-images.githubusercontent.com/5810477/105111944-01fbc600-5a90-11eb-9fff-cfc9b042fbfb.png)|![more-enabled-android](https://user-images.githubusercontent.com/5810477/105111960-07f1a700-5a90-11eb-86ee-0ef15ea5bd07.png)|![more-disabled-android](https://user-images.githubusercontent.com/5810477/105111962-09bb6a80-5a90-11eb-86ae-db2c18db55b7.png)






## Types of changes
- Added new `.disabled` style for the `inserter-button` component
- Passed the `item.isDisabled` value to the UI (this value already existed)

## Checklist:
